Update Examples to the lastest Rust nightly

bitborn opened the pull request:
I updated the examples and tests to at very least build with the latest Rust nightly (build 2015-03-31). Hopefully this will let get the tests passing.

Obviously there's a lot more work to do, but I think just getting it to compile is a good start.

kytrinyx commented:
Oh, my, a passing build :) It's been a while!

@pminten, @etrepum I'll leave this to your discretion.

pminten commented on parallel-letter-frequency/example.rs:
```diff
@@ -41,9 +40,9 @@ pub fn frequency(texts: &[&str], num_workers: usize) -> HashMap<char, usize> {
+        for c in line.chars() {
             if c.is_alphabetic() {
-                match results.entry(c.to_lowercase()) {
+                match results.entry(c.to_ascii_lowercase()) {
```
The test contains some non-7-bit-ASCII characters. I'm sure the tests pass with this but I'm not sure that's by accident or because it does the right thing.

bitborn replied:
I believe the [ASCII lowercase](http://doc.rust-lang.org/std/ascii/trait.AsciiExt.html#tymethod.to_ascii_lowercase) method does a check to see if the char is actually ASCII. Though if we included a test with both a upper and lower case Unicode character (example: "Θ θ"), I believe it would count these as separate characters...

I will update using the [Unicode version](http://doc.rust-lang.org/std/primitive.char.html#method.to_lowercase) which should, if I read the docs correctly, count Unicode upper and lower ("Θ θ") as the same character.

pminten commented:
Looks good to me, though I don't have enough experience to do give anything but the most cursory review.

bitborn commented:
Just added the Unicode method in the parallel frequency example. 

I think there may be an edge case we need to worry about down the road. Basically, in future versions the `to_lowercase` method may return an iterator of more than one character (which I believe is called case folding). Theres a little blurb about it in the docs for [`to_uppercase`](http://doc.rust-lang.org/std/primitive.char.html#method.to_uppercase).

I also updated this PR to the latest nightly.

pminten commented:
I wouldn't worry about case folding returning multiple codepoints, that should be pretty rare outside pathological constructed cases.

The rest looks good. Since nobody else has commented I'll just merge this.

pminten commented:
Merged.

Thanks for the massive work.
